### ISSUE TITLE
Cloudinary - File upload extension

### ISSUE THREAD

**Issue by bejoinka** [awell-health/awell-extensions#87]
# File upload extensions
These extensions allow users in hosted pages to upload a file to a cloud resource / bucket. The uri and other information about the upload should be passed as data points once the upload is complete.

## Cloudinary
Nick began work on the Cloudinary extension, which supports file uploads. However, after some time this weekend, it wasn't immediately clear what was going on. There is already already some work in awell-extensions to handle, but we're going to need to do some work in hosted pages to support

Currently, the extension has some code here to support the extension vis-a-vis settings, but it's going to take some work in [hosted pages](https://github.com/awell-health/hosted-pages) to complete.

The extension has already been merged into main in order to support testing in hosted pages.

Could we timebox an effort here to try to get something up and running? How is one day?

**Comment by bejoinka** [comment]
If we're unable to get a POC together with cloudinary, perhaps we park it in the backlog. If, however, we are able to get something up and running, we're probably going to need some design support from @nckhell vis-a-vis the hosted pages, and we still need to spend some time deciding what kind of data we want to pass as fields / datapoints.

**Comment by nckhell** [comment]
Actually, the code for Awell Hosted Pages is 80% done I believe. I just wasn't able to test it as I had some issues connecting my local environment to the development environment. So that PR is probably a good place to start.

https://github.com/awell-health/hosted-pages/pull/116

**Comment by nckhell** [comment]
> The uri and other information about the upload should be passed as data points once the upload is complete.

I think that for a V1/POC I would already be more than happy if we can just upload files without passing any information back to the care flow. To make sure these files can easily be retrieved later, I was planning to use the `context` object which you can pass when uploading a file (see API reference [here](https://cloudinary.com/documentation/upload_widget_reference)) + work with tags which the user can set in Awell Studio.

<img width="869" alt="Screenshot 2023-04-18 at 08 52 32" src="https://user-images.githubusercontent.com/5594064/232695256-2ff7b68f-8fcc-4cc9-b84c-b0a5a382f4aa.png">

<img width="868" alt="Screenshot 2023-04-18 at 08 52 47" src="https://user-images.githubusercontent.com/5594064/232695316-d4c45f36-7fdc-4e0a-b931-e9b489ae6f08.png">

```json
{
  "context": {
    "awellPatientId": "ABC", // the id of the patient the file was uploaded for
    "awellPathwayId": "DEF", // the id of the pathway the file was uploaded for
    "awellActivityId": "GHI", // the id of the activity that uploaded the file(s)
  }
}
```

**Comment by nckhell** [comment]
> We're probably going to need some design support from @nckhell vis-a-vis the hosted pages, and we still need to spend some time deciding what kind of data we want to pass as fields / datapoints.

@michal-grzelak In terms of how the activity is rendered in Awell Hosted Pages - you can keep it simple for now and come up with something yourself. I assume what we minimally need is an "upload" button and a "submit" button. I am much more interested in getting it to work first, improving how it looks is a rather trivial thing to do after that.

cc @bejoinka
